src/Pages/Shared/ConfigMapSecret/ConfigMapSecret.container.tsx

```diff
@@ -110,60 +118,91 @@ export const ConfigMapSecretContainer = ({
+                                      ? ConfigResourceType.ConfigMap
+                                      : ConfigResourceType.Secret,
+                          },
+                          abortControllerRef.current.signal,
+                      ),
             ])
         }, abortControllerRef)
-            .then(([_draftData, _cmSecretData]) => {
+            .then(([draftDataRes, cmSecretDataRes]) => {
                 let draftId: number
                 let draftState: number
-                let _configMap: ConfigMapSecretData
+                let _configMapSecret: ConfigMapSecretData
 
                 if (
-                    _draftData.status === 'fulfilled' &&
-                    _draftData.value?.result &&
-                    (_draftData.value.result.draftState === DraftState.Init ||
-                        _draftData.value.result.draftState === DraftState.AwaitApproval)
+                    draftDataRes.status === 'fulfilled' &&
+                    draftDataRes.value?.result &&
+                    (draftDataRes.value.result.draftState === DraftState.Init ||
+                        draftDataRes.value.result.draftState === DraftState.AwaitApproval)
                 ) {
                     setDraftData({
-                        ..._draftData.value.result,
-                        unAuthorized: _draftData.value.result.dataEncrypted,
+                        ...draftDataRes.value.result,
+                        unAuthorized: draftDataRes.value.result.dataEncrypted,
                     })
-                    draftId = _draftData.value.result.draftId
-                    draftState = _draftData.value.result.draftState
+                    draftId = draftDataRes.value.result.draftId
+                    draftState = draftDataRes.value.result.draftState
 
                     setSelectedTab(
-                        _draftData.value.result.draftState === DraftState.AwaitApproval
+                        draftDataRes.value.result.draftState === DraftState.AwaitApproval
                             ? CMSecretProtectedTab.Compare
                             : CMSecretProtectedTab.Draft,
                     )
                 } else {
                     setDraftData(null)
                 }
 
-                if (_cmSecretData.status === 'fulfilled') {
-                    if (_cmSecretData.value?.result?.configData?.length) {
-                        _configMap = {
-                            ..._cmSecretData.value.result,
+                if (cmSecretDataRes.status === 'fulfilled') {
+                    const _cmSecretData = isJob
+                        ? cmSecretDataRes.value?.result
+                        : cmSecretDataRes.value?.result[
+                              componentType === CMSecretComponentType.ConfigMap ? 'configMapData' : 'secretsData'
+                          ].data
+
+                    const unAuthorized = isJob
+                        ? false
```

Comment:
why are we hardcoding in case of job?

Reply:
currently job does not have support for non-admin users
added comment